functests/1_performance/netqueues.go

```diff
@@ -0,0 +1,359 @@
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/cluster"
+	testlog "github.com/openshift-kni/performance-addon-operators/functests/utils/log"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+)
+
+var _ = Describe("[ref_id: 40307][pao] updating performance profile", func() {
+	var workerRTNodes []corev1.Node
+	var err error
+	var net bool = true
+	var profile *performancev2.PerformanceProfile
+	testutils.BeforeAll(func() {
+		isSNO, err := cluster.IsSingleNode()
+		Expect(err).ToNot(HaveOccurred())
+		RunningOnSingleNode = isSNO
+	})
+	BeforeEach(func() {
+		workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred())
+		nodeLabel := testutils.NodeSelectorLabels
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("Updating performance profile for netqueues", func() {
+		It("[test_id:40308][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Network device queues Should be set to the profile's reserved CPUs count ", func() {
+			devices := make(map[string]int)
+			count := 0
+			if profile.Spec.Net != nil {
+				if profile.Spec.Net.UserLevelNetworking != nil && *profile.Spec.Net.UserLevelNetworking && len(profile.Spec.Net.Devices) == 0 {
+					By("To all non virtual network devices when no devices are specified under profile.Spec.Net.Devices")
+					err := checkDeviceSupport(workerRTNodes, devices)
+					Expect(err).ToNot(HaveOccurred())
+					for _, size := range devices {
+						if size == getReservedCPUSize(profile.Spec.CPU) {
+							count++
+						}
+					}
+					if count == 0 {
+						Skip("Skipping Test: Unable to set Network queue size to reserved cpu count")
+					}
+				}
+			}
+		})
+
+		It("[test_id:40542] Verify the number of network queues of all supported network interfaces are equal to reserved cpus count", func() {
+			devices := make(map[string]int)
+			count := 0
+			if profile.Spec.Net == nil {
+				By("Enable UserLevelNetworking in Profile")
+				profile.Spec.Net = &performancev2.Net{
+					UserLevelNetworking: &net,
+				}
+				profiles.UpdateWithRetry(profile)
+			}
+			// Populate the device map with queue sizes
+			Eventually(func() bool {
+				err := checkDeviceSupport(workerRTNodes, devices)
+				Expect(err).ToNot(HaveOccurred())
+				return true
+			}, cluster.ComputeTestTimeout(200*time.Second, RunningOnSingleNode), testPollInterval*time.Second).Should(BeTrue())
+			for _, size := range devices {
```

Comment:
same comment as above on why are we using count and not throwing an error in case of a mismatch

Reply:
@yanirq the reason is that on the upstream GCP RT lane, the hardware doesn't seem to support resize and so currently we are skipping in case failure .

Reply:
ack